waku/v1/protocol/waku_protocol.nim

```diff
@@ -171,7 +171,7 @@ proc read*(rlp: var Rlp, T: typedesc[StatusOptions]): T =
     of bloomFilterKey:
       let bloom = rlp.read(seq[byte])
       if bloom.len != bloomSize:
-        raise newException(UselessPeerError, "Bloomfilter size mismatch")
+        raise newException(RlpError, "Bloomfilter size mismatch")
```

Comment:
Yeah that is fine. It should have been an `RlpError` to begin with as it is RLP decoding of `StatusOptions` (even though that size check is not technically an RlpError I guess).

Though perhaps you better use `RlpTypeMismatch` instead of the general `RlpError`.

Reply:
Thanks, @kdeme